internal/provider/source_v2/source_v2_schemas_data_source.go

```diff
@@ -0,0 +1,160 @@
+package source_v2
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"terraform-provider-sifflet/internal/apiclients"
+	sifflet "terraform-provider-sifflet/internal/client"
+	"terraform-provider-sifflet/internal/tfutils"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	_ datasource.DataSource              = &sourceV2SchemasDataSource{}
+	_ datasource.DataSourceWithConfigure = &sourceV2SchemasDataSource{}
+)
+
+func newSourceV2SchemasDataSource() datasource.DataSource {
+	return &sourceV2SchemasDataSource{}
+}
+
+type sourceV2SchemasDataSource struct {
+	client *sifflet.ClientWithResponses
+}
+
+func (d *sourceV2SchemasDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	clients, ok := req.ProviderData.(*apiclients.HttpClients)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *HttpClients, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = clients.Client
+}
+
+func (d *sourceV2SchemasDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_source_v2_schemas"
+}
+
+func SourceV2SchemasDataSourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Description: `Get all schemas for a given source.
+
+"Schemas" here is used broadly to refer to all underlying schemas/workspaces/databases/instances contained in a Sifflet source.
+
+This will return the URIs of the schemas for the source. Read more about URIs here: https://docs.siffletdata.com/docs/uri.
+
+For example:
+- For an Airflow source, this will return the URIs of the Airflow instances contained in the source: https://docs.siffletdata.com/docs/airflow-uri-format.
+- For an Athena source, this will return the URIs of the databases contained in the source: https://docs.siffletdata.com/docs/athena-uri-format.
+- For a MySQL source, this will return the URIs of the schemas contained in the source: https://docs.siffletdata.com/docs/mysql-uri-format.
+
+~> Note that the schemas are not immediately available after the source is created. You need to wait for the initial source ingestion to finish before the schemas are available. As such, we do not recomment using this data source in the same module as the one that creates the corresponding Sifflet source.`,
```

Comment:
```suggestion
~> Note that the schemas are not immediately available after the Sifflet source is created. You need to wait for the initial Sifflet source run to finish before the schemas are available. As such, we do not recomment using this data source in the same module as the one that creates the corresponding Sifflet source.`,
```

* let's be explicit about Sifflet source vs Terraform data source (it can be confusing otherwise)
* the Sifflet UI never uses "ingestion", I think that's a technical term only. I suggested "run" above to mirror the UI, but feel free to disagree and use any other term (or keep "ingestion")